.github/workflows/checks.yml

```diff
@@ -7,7 +7,7 @@ on:
   workflow_dispatch:
 
 env:
-  GO_VERSION: "1.20"
+  GO_VERSION: "~1.21.1"
```

Comment:
```suggestion
  GO_VERSION: "1.21"
```

Reply:
Why not set `1.21.1` as the minimum here, `1.21.0` includes security issues resolved by `1.21.1`?

Reply:
`1.21` is not `1.21.0`. If you set `check-latest: true` it should grab the latest `1.21.x` version.
See https://github.com/actions/setup-go#check-latest-version

Reply:
[It should "grab the latest `1.21.x` version" with `"~1.21.1"` as well](https://github.com/npm/node-semver): 

![20230907_070134](https://github.com/open-telemetry/opentelemetry-go-instrumentation/assets/5543599/9dbef148-5fd7-4610-8181-295396ce2a7a)

However, this would ensure `1.21.0` is never used.

Reply:
Yeah, I can see adding `check-latest: true`. Just wondering if this needed.

Reply:
> However, this would ensure 1.21.0 is never used.

Agreed, and this was my intent through the waves of PRs I submitted yesterday. I think the combination of both `go-version` and `check-latest` is the safest way to ensure that a minimum versions is used.